src/gdbstub_impl/mod.rs

```diff
@@ -23,6 +23,32 @@ pub use error::GdbStubError;
 
 use GdbStubError as Error;
 
+/// Describes the offset the target loaded the image sections at so the target
+/// can notify GDB that it needs to adjust the addresses of symbols
+pub enum SectionOffsets<U> {
```

Comment:
Lets move this struct into `crate::target::ext::offsets`, since it's only relevant for those implementing the `SectionOffsets` extension.